api/src/main/java/jakarta/persistence/EntityHandler.java

```diff
@@ -0,0 +1,529 @@
+     *         fail)
+     * @since 2.1
+     */
+    StoredProcedureQuery createStoredProcedureQuery(
+            String procedureName, Class<?>... resultClasses);
+
+    /**
+     * Create an instance of {@link StoredProcedureQuery} for executing
+     * a stored procedure in the database.
+     * <p>Parameters must be registered before the stored procedure can
+     * be executed.
+     * <p>The {@code resultSetMapping} arguments must be specified in
+     * the order in which the result sets is returned by the stored
+     * procedure invocation.
+     * @param procedureName name of the stored procedure in the
+     *        database
+     * @param resultSetMappings the names of the result set mappings
+     *        to be used in mapping result sets
+     *        returned by the stored procedure
+     * @return the new stored procedure query instance
+     * @throws IllegalArgumentException if a stored procedure or
+     *         result set mapping of the given name does not exist
+     *         (or the query execution will fail)
+     */
+    StoredProcedureQuery createStoredProcedureQuery(
+            String procedureName, String... resultSetMappings);
+
+    /**
+     * Join the current active JTA transaction.
+     * <p>This method should be called on a JTA application-managed
+     * {@code EntityHandler} that was created outside the scope of
+     * the active transaction or on an {@code EntityHandler} of
+     * type {@link SynchronizationType#UNSYNCHRONIZED} to associate
+     * it with the current JTA transaction.
+     * @throws TransactionRequiredException if there is no active
+     *         transaction
+     */
+    void joinTransaction();
+
+    /**
+     * Determine whether the {@code EntityHandler} is joined to the
+     * current transaction. Returns false if the {@code EntityHandler}
+     * is not joined to the current transaction or if no
+     * transaction is active.
+     * @return boolean
+     * @since 2.1
+     */
+    boolean isJoinedToTransaction();
+
+    /**
+     * Return an object of the specified type to allow access to
+     * a provider-specific API. If the provider implementation
+     * of {@code EntityHandler} does not support the given type,
+     * the {@link PersistenceException} is thrown.
+     * @param cls  the class of the object to be returned.
+     *             This is usually either the underlying class
+     *             implementing {@code EntityHandler} or an
+     *             interface it implements.
+     * @return an instance of the specified class
+     * @throws PersistenceException if the provider does not
+     *         support the given type
+     * @since 2.0
+     */
+    <T> T unwrap(Class<T> cls);
+
+    /**
+     * Return the underlying provider object for the
+     * {@link EntityHandler}, if available. The result of this
+     * method is implementation-specific.
+     * <p>The {@code unwrap} method is to be preferred for new
+     * applications.
```

Comment:
maybe let's deprecate it then and don't bring it to the common interface ?

Reply:
I agree with omitting `getDelegate` from the common interface.